manifests/ssl_bump.pp

```diff
@@ -1,41 +1,29 @@
-# @summary 
+# @summary
 #   Defines ssl_bump entries for a squid server.
 # @see
-#   http://www.squid-cache.org/Doc/config/ssl_bump/ 
+#   http://www.squid-cache.org/Doc/config/ssl_bump/
 # @example
 #   squid::ssl_bump { 'all':
 #     action => 'bump',
 #   }
-# 
+#
 #   Adds a squid.conf line
 #   ssl_bump bump all
-# 
-# @param title 
+#
+# @param title
 #   The name of acl the ssl_bump rule is applied to
-# @param action 
+# @param action
 #   The type of the ssl_bump, must be defined, e.g bump, peek, ..
 # @param order
 #   Order can be used to configure where in `squid.conf`this configuration section should occur.
 define squid::ssl_bump (
-  Enum[
-    'bump',
-    'client-first',
-    'none',
-    'peek',
-    'peek-and-splice',
-    'server-first',
-    'splice',
-    'stare',
-    'terminate']
-          $action = 'bump',
-  String  $value  = $title,
-  String  $order  = '05',
+  Squid::Action $action = 'bump',
```

Comment:
Action can be ambiguous. If you look at it, it's most often `Enum['allow', 'deny']`. Perhaps we should have `Squid::Action::AllowDeny` and `Squid::Action::SslBump`.

Reply:
I think we should do this in another PR?

Reply:
You are introducing the type in this PR so I'd prefer to at least do it right the first time for any newly introduced type (when we can). Not asking to introduce `Squid::Action::AllowDeny` now, just to name it `Squid::Action::SslBump`